File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = Counter(nums)
        bucketSort = [[] for _ in range(len(nums))]
        for num, count in counter.items():
            bucketSort[count-1].append(num)
        res = []
        tracker = 0
        for i in range(len(bucketSort)-1, -1, -1):
            if not bucketSort[i]:
                continue;
            for item in bucketSort[i]:
                res.append(item)
                tracker+=1
                if tracker == k:
                    break
            if tracker == k:
                break
        return res

File: Data Structures & Algorithms/palindromic-substrings/submission-2.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count = 0;
        for (int i = 0; i < s.size(); ++i) {
            for (int offset = 0; offset < 2; ++offset) {
                int l = i, r = i + offset;
                while (l >= 0 && r < s.size()) {
                    if (s[l] != s[r]) {
                        break;
                    } else {
                        count++;
                        l--;
                        r++;
                    }
                }
            }
        }
        return count;
    }
};
